app/uk/gov/hmrc/customs/notification/services/CustomsNotificationService.scala

```diff
@@ -103,11 +103,18 @@ class CustomsNotificationService @Inject()(logger: NotificationLogger,
     pushOrPullService.send(workItem.item, apiSubscriptionFields).map {
       case Right(connector) =>
         notificationWorkItemRepo.setCompletedStatus(workItem.id, Succeeded)
-        logger.info(s"$connector ${Succeeded.name} for workItemId ${workItem.id.toString}")
+        val start = connector match {
+          case Push => s"$connector ${Succeeded.name}"
+          case Pull => s"Placing on $connector queue ${Succeeded.name}"
+        }
+        logger.info(start + s" for workItemId ${workItem.id.toString}")
         true
       case Left(pushOrPullError) =>
-        val msg = s"${pushOrPullError.source} failed ${pushOrPullError.toString} for workItemId ${workItem.id.toString}"
-        logger.warn(msg)
+        val start = pushOrPullError.source match {
+          case Push => s"${pushOrPullError.source} failed ${pushOrPullError.toString}"
+          case Pull => s"Placing on ${pushOrPullError.source} queue failed ${pushOrPullError.toString}"
```

Comment:
You may need a catch-all error option as it looks like GetApiSubscriptionFields could be another place to error, also catch-alls ensure we get logs for anything out of scope.

Reply:
ConnectorSource is as sealed trait, and can only be Push or Pull.

Would this be the catch all you mean?

from line 139.
    }.recover {
      case NonFatal(e) =>
        logger.error(s"failed push/pulling notification work item with csid: ${workItem.item._id.toString} and conversationId: ${workItem.item.notification.conversationId.toString} due to: $e")
        false
    }

Reply:
Yes thanks, sorry, my mistake there are three after all.
```
sealed trait ConnectorSource
case object Push extends ConnectorSource
case object Pull extends ConnectorSource
case object GetApiSubscriptionFields extends ConnectorSource
```